### PR TITLE
Directory management

### PR DESCRIPTION
**Pull request by jsteel44**

### CODE REVIEW

File: internal/pkg/filesystem_impl/mount.go [hunk 1]
```diff
@@ -229,8 +213,8 @@ func mountBeegFS(hostname string, mgtHost string, fsname string, directory strin
 	return runner.Execute(hostname, fmt.Sprintf("ln -s /mnt/beegfs/%s %s", fsname, directory))
 }
 
-func mkdir(hostname string, directory string) error {
-	return runner.Execute(hostname, fmt.Sprintf("mkdir -p %s", directory))
+func mkdir(hostname string, owner uint, group uint, directory string) error {
```
Comment (JohnGarbutt) [inline internal/pkg/filesystem_impl/mount.go]: I think we should rename this to install

**Comment by jsteel44** [comment]
OK I'll hold off on the install/mkdir one for now until a later date, thanks.

File: internal/pkg/filesystem_impl/mount.go [hunk 2]
```diff
@@ -69,9 +66,6 @@ func mount(fsType FSType, sessionName datamodel.SessionName, isMultiJob bool, in
 			if err := mkdir(attachHost, privateDir); err != nil {
 				return err
 			}
-			if err := fixUpOwnership(attachHost, owner, group, privateDir); err != nil {
```
Comment (JohnGarbutt) [inline internal/pkg/filesystem_impl/mount.go]: I am planning to move the creation of these directories so it only happens on the primary brick host during create. However in that version it means we still need the mkdir here, instead the private directoy and global directory will get the permissions set spearately.

Now that is a thing we should do with install.

So maybe we should change this more, i.e.:
* add create of the /private and /global directories before this larget mount loop.
* only need to do the install on one host really, so maybe just pick the first host and do it there?

I might try do a bit of this rework myself it help kick this off...

Comment (jsteel44) [reply internal/pkg/filesystem_impl/mount.go]: Sorry I missed this comment before closing and re-creating the pull request. Would it make sense for you to pull in my branch and continue working on it? I'll leave it with you, thanks.

**Pull request closed without merging**